### PR TITLE
Add Git CLI task

### PR DESCRIPTION
**Pull request by Divyansh42**
# Changes

Added `Git CLI` task, this task can be used to run `git commands`.
Git command need to be passed as a `script` to the task

Signed-off-by: Divyansh42 <diagrawa@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: git/examples/git-cli/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      workspaces:
+        - name: source
+          workspace: shared-workspace
+        - name: input
+          workspace: input
+      params:
+        - name: GIT_USER_NAME
+          value: "divyansh42"
+        - name: GIT_USER_EMAIL
+          value: diagrwa@redhat.com
```
[the remark this answers is not in the document]
Comment (Divyansh42) [reply git/examples/git-cli/pipeline.yaml:40]: Okay :sweat_smile:

File: git/examples/git-cli/pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      runAfter:
+        - fetch-repository
+      workspaces:
+        - name: source
+          workspace: shared-workspace
+        - name: input
+          workspace: input
+      params:
+        - name: GIT_USER_NAME
+          value: $(username)
```
Comment (savitaashture) [inline git/examples/git-cli/pipeline.yaml:36]: `$(username)` is confusing here and indicates that the value filled somewhere and you are using here

So can we change something like below 
```suggestion
          value: username
```
OR
```suggestion
          value: username-name
```

Comment (Divyansh42) [reply git/examples/git-cli/pipeline.yaml:36]: We can, thanks for the suggestion :)

**Comment by savitaashture** [comment]
/lgtm

Thank you :+1:

**Pull request merged**